stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/MessageSendingService.kt

```diff
@@ -65,15 +65,23 @@ internal class MessageSendingService(
     }
 
     internal suspend fun sendMessage(message: Message): Result<Message> {
-        return if (domainImpl.online.value) {
-            return if (message.hasPendingAttachments()) {
-                waitForAttachmentsToBeSent(message)
-            } else {
-                doSend(message)
+        return when {
+            domainImpl.online.value ->
+                if (message.hasPendingAttachments()) {
+                    waitForAttachmentsToBeSent(message)
+                } else {
+                    doSend(message)
+                }
+
+            message.hasPendingAttachments() -> {
```

Comment:
I think if we're offline we don't need to do anything at all. Just return what we have in else branch. Because when connection recovered we have handling logic already for this

Reply:
If we do this, then we won't enqueue the attachments and it won't be sent in the background. Basically the ticket won't be done

Reply:
Our logic for recovered connection only works if the app is in the foreground as far as I know. 🤔  If that's what we want, then there's no need to merge this PR

Reply:
`domainImpl.online` won't be `true` if WebSocket is disconnected and we're not connecting it while the app is in the background. Maybe we should distinguish between no internet connection and socket disconnected somehow 🤔

Reply:
Yes, it can work! But for then to exclude double sending, before to start send attachments we need to update their status in DB for `InProgress`

Reply:
I believe this is already happening here:
```
.onSuccessSuspend { newMessage ->
            // we insert early to ensure we don't lose messages
            domainImpl.repos.insertMessage(newMessage)
            domainImpl.repos.updateLastMessageForChannel(newMessage.cid, newMessage)
        }
```